packages/client/tests/functional/json-null-types/tests.ts

```diff
@@ -0,0 +1,119 @@
+        const data = await prisma.nullableJsonField.create({
+          data: {
+            json: Prisma.DbNull,
+          },
+        })
+        expect(data.json).toBe(null)
+      })
+    })
+
+    describe('requiredJsonField', () => {
+      test('JsonNull', async () => {
+        const data = await prisma.requiredJsonField.create({
+          data: {
+            json: Prisma.JsonNull,
+          },
+        })
+        expect(data.json).toBe(null)
+      })
+
+      test('DbNull', async () => {
+        await expect(
+          prisma.requiredJsonField.create({
+            data: {
+              // @ts-expect-error
+              json: Prisma.DbNull,
+            },
+          }),
+        ).rejects.toThrowErrorMatchingInlineSnapshot(`
+
+                Invalid \`prisma.requiredJsonField.create()\` invocation in
+                /client/tests/functional/json-null-types/tests.ts:42:36
+
+                  39 
+                  40 test('DbNull', async () => {
+                  41   await expect(
+                → 42     prisma.requiredJsonField.create({
+                           data: {
+                             json: Prisma.DbNull
+                                   ~~~~~~~~~~~~~
+                           }
+                         })
+
+                Argument json: Provided value Prisma.DbNull of type DbNull on prisma.createOneRequiredJsonField is not a enumTypes.
+                → Possible values: JsonNullValueInput.JsonNull
+
+
+              `)
+      })
+    })
+
+    describe('properties of DbNull/JsonNull/AnyNull', () => {
+      test('instanceof checks pass', () => {
+        expect(Prisma.DbNull).toBeInstanceOf(Prisma.NullTypes.DbNull)
+        expect(Prisma.JsonNull).toBeInstanceOf(Prisma.NullTypes.JsonNull)
+        expect(Prisma.AnyNull).toBeInstanceOf(Prisma.NullTypes.AnyNull)
+      })
+
+      test('custom instances are not allowed', async () => {
+        await expect(
+          prisma.requiredJsonField.create({
+            data: {
+              // @ts-expect-error
+              json: new Prisma.NullTypes.JsonNull(),
+            },
+          }),
+        ).rejects.toThrowErrorMatchingInlineSnapshot(`
+
+                Invalid \`prisma.requiredJsonField.create()\` invocation in
+                /client/tests/functional/json-null-types/tests.ts:80:36
+
+                   77 
+                   78 test('custom instances are not allowed', async () => {
+                   79   await expect(
+                →  80     prisma.requiredJsonField.create({
+                            data: {
+                              json: new Prisma.NullTypes.JsonNull()
+                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+                            }
+                          })
+
+                Argument json: Provided value new Prisma.NullTypes.JsonNull() of type JsonNull on prisma.createOneRequiredJsonField is not a enumTypes.
+                → Possible values: JsonNullValueInput.JsonNull
```

Comment:
Should it be `Prisma.JsonNull` here?

Reply:
Well, good question.

On the one hand, it would've been nice UX, but, on the other hand, it might be pretty complicated to change and inconsistent with validation errors that involve other enums (they are always printed as `Enum.Variant`).

FWIW, using `Prisma.JsonNullValueInput.JsonNull` is totally valid and identical to just `Prisma.JsonNull`.

Reply:
So, I took a look, and fixed a bug on the line above while I was there (should have said `... is not a JsonNullValueInput`, but due to a typo it printed `bestFittingType.location` instead of `bestFittingType.type` and said `... is not a enumTypes` — now fixed for all enums).

It looks like it shouldn't be hard to add a special case there, but I would prefer the output to be consistent for all kinds of enums:
* either `JsonNullValueInput.JsonNull`, `SortOrder.asc` etc (like it is now)
* or `Prisma.JsonNull`, `'asc'` etc (we can change it to be like this)

What do you think? Or should we discuss and possibly change this later?

Reply:
Thanks for fixing the bug! Let's do that later and merge this. Feel free to open an improvement/DX issue, but I don't think it's very urgent right now.